drivers/place/attendee_scanner.cr

```diff
@@ -0,0 +1,49 @@
+require "placeos-driver"
+require "json"
+
+class Place::AttendeeScanner < PlaceOS::Driver
+  descriptive_name "PlaceOS Room Events"
+  generic_name :AttendeeScanner
+
+  getter systems : Hash(String, Array(String)) { get_systems_list.not_nil! }
+
+  # Grabs the list of systems in the building
+  def get_systems_list
+    building_id = system[:Location].building_id.get
+    system["StaffAPI"].systems_in_building(building_id).get.as_h.transform_values(&.as_a.map(&.as_s))
+  rescue error
+    logger.warn(exception: error) { "unable to obtain list of systems in the building" }
+    nil
+  end
+
+  # Lists external guests based on email domains
+  def list_external_guests
+    guests = [] of Hash(String, JSON::Any)
+
+    systems.each do |level_id, system_ids|
+      system_ids.each do |system_id|
+        sys = system(system_id)
+        if sys.exists?(:Booking)
+          bookings = sys[:Booking][:bookings].as_a
+
+          bookings.map do |booking|
+            unless booking["host"].as_s.includes?("place.technology")
```

Comment:
There will almost never be a host that matches this.
We'd need a setting that allows for an array of internal to the organisation domain names `["pwc.com.au", "pwc.com"]` etc then we'd want to check if there are any attendee emails that don't match these domains

Reply:
>filtering out any that are internal staff (email domains)

This confused me and I thought it would be simply place.technology, that can be arranged.